public/app/features/query/components/QueryGroup.tsx

```diff
@@ -130,7 +130,10 @@ export class QueryGroup extends PureComponent<Props, State> {
   }
 
   onPanelDataUpdate(data: PanelData) {
-    this.setState({ data });
+    this.setState({
+      queries: this.props.options.queries,
+      data,
+    });
   }
```

Comment:
@torkelo I removed the specific Datagrid event from QueryGroup and instead publish a `PanelQueriesChangedEvent`, which updates the PanelEditorTabs. Problem is that when data changes and updates the QueryGroup it doesn't update the queries and thus doesn't update the UI showing that a snapshot is in use. This line fixes that, wdyt?

Using the PanelContext didn't help much because I need to specifically set the state in the queryGroup.

Reply:
@mdvictor I think this could clear the defaults that are applied onmount

Reply:
@mdvictor If I debug this clearly undoes the logic in onmount where query defaults are applied to the initial empty query model
https://github.com/grafana/grafana/blob/main/public/app/features/query/components/QueryGroup.tsx#L103

Reply:
@torkelo You are right. I've changed the code to update the state properly. wdyt? I added a `componentDidUpdate` so that when the `PanelEditorTabs` get updated by the `PanelQueriesChangedEvent` it also triggers a state update in `QueryGroup`

Reply:
@mdvictor should work, but quite a bit of duplication now between mount and on update code. Would be good to extract that to a shared function